src/mad_spark_alt/core/timeout_wrapper.py

```
@@ -0,0 +1,170 @@
+    
+    def __init__(self, message: str, phase: Optional[str] = None, elapsed: Optional[float] = None):
+        self.phase = phase
+        self.elapsed = elapsed
+        super().__init__(message)
+
+
+async def with_timeout(
+    coro: Callable[..., Any],
+    timeout: float,
+    phase: str = "unknown",
+    fallback: Optional[Any] = None,
```

Ruff (0.11.9)

Dynamically typed expressions (typing.Any) are disallowed in `fallback`

(ANN401)